builder/vsphere/supervisor/step_import_image.go

```diff
@@ -0,0 +1,378 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+//go:generate packer-sdc struct-markdown
+//go:generate packer-sdc mapstructure-to-hcl2 -type ImportImageConfig
+
+package supervisor
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/hashicorp/packer-plugin-sdk/multistep"
+	imgregv1 "github.com/vmware-tanzu/image-registry-operator-api/api/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/fields"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	ImportTargetKind       = "ContentLibrary"
+	ImportTargetAPIVersion = "imageregistry.vmware.com/v1alpha1"
+
+	DefaultWatchImageImportTimeoutSec = 600
+
+	ImportRequestDefaultNamePrefix = "packer-vsphere-supervisor-import-req-"
+
+	StateKeyImageImportRequestCreated = "item_import_req_created"
+	StateKeyImportedImageName         = "imported_image_name"
+
+	importOVFFeatureNotEnabledMsg = "WCP_VMImageService_ImportOVF feature is not enabled"
+)
+
+var IsWatchingImageImport bool
+
+type ImportImageConfig struct {
+	// The remote URL where the to-be-imported image is hosted.
+	ImportSourceURL string `mapstructure:"import_source_url"`
+	// The SSL certificate of the remote HTTP server that hosts the to-be-imported image.
+	ImportSourceSSLCertificate string `mapstructure:"import_source_ssl_certificate"`
+	// Name of a writable and import-allowed ContentLibrary resource in the namespace where the image will be imported.
+	ImportTargetLocationName string `mapstructure:"import_target_location_name"`
+	// The type of imported image.
+	// Defaults to `ovf`. Available options include `ovf`.
+	ImportTargetImageType string `mapstructure:"import_target_image_type"`
+	// Name of the imported image.
+	// Defaults to the file name of the image referenced in the source URL.
+	ImportTargetImageName string `mapstructure:"import_target_image_name"`
+	// The name of the image import request.
+	// Defaults to `packer-vsphere-supervisor-import-req-<random-suffix>`.
+	ImportRequestName string `mapstructure:"import_request_name"`
+	// The timeout in seconds to wait for the image to be imported.
+	// Defaults to `600`.
+	WatchImportTimeoutSec int `mapstructure:"watch_import_timeout_sec"`
+	// Preserve the import request in the Supervisor cluster after the build finishes.
+	// Defaults to `false`.
+	KeepImportRequest bool `mapstructure:"keep_import_request"`
+	// Clean the imported image after the build finishes. If set to `true`, the imported image will be deleted.
+	// Defaults to `false`.
+	CleanImportedImage bool `mapstructure:"clean_imported_image"`
+}
+
+func (c *ImportImageConfig) Prepare() []error {
+	if c.ImportSourceURL == "" {
+		return nil
+	}
+
+	var errs []error
+	if c.ImportSourceURL != "" && c.ImportTargetLocationName == "" {
```

Comment:
The first condition seems unnecessary as the function returns if it's empty above.

Reply:
Yep, it can be removed.